app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/localService.ts

```diff
@@ -0,0 +1,157 @@
+            const project = result.projects.find(listedProject => listedProject.id === projectId)
+            const engineVersion = project?.engineVersion
+            if (project == null) {
+                throw new Error(`The project ID '${projectId}' is invalid.`)
+            } else if (engineVersion == null) {
+                throw new Error(`The project '${projectId}' does not have an engine version.`)
+            } else {
+                return Promise.resolve<cloudService.Project>({
+                    name: project.name,
+                    engineVersion: {
+                        lifecycle: cloudService.VersionLifecycle.stable,
+                        value: engineVersion,
+                    },
+                    ideVersion: {
+                        lifecycle: cloudService.VersionLifecycle.stable,
+                        value: engineVersion,
+                    },
+                    jsonAddress: null,
+                    binaryAddress: null,
+                    organizationId: '',
+                    packageName: project.name,
+                    projectId,
+                    state: {
+                        type:
+                            projectId === this.currentlyOpeningProjectId
+                                ? cloudService.ProjectState.openInProgress
+                                : cloudService.ProjectState.closed,
+                    },
+                })
+            }
+        } else {
+            const project = this.currentlyOpenProject.project
+            return Promise.resolve<cloudService.Project>({
+                name: project.projectName,
+                engineVersion: {
+                    lifecycle: cloudService.VersionLifecycle.stable,
+                    value: project.engineVersion,
+                },
+                ideVersion: {
+                    lifecycle: cloudService.VersionLifecycle.stable,
+                    value: project.engineVersion,
+                },
+                jsonAddress: ipWithSocketToAddress(project.languageServerJsonAddress),
+                binaryAddress: ipWithSocketToAddress(project.languageServerBinaryAddress),
+                organizationId: '',
+                packageName: project.projectName,
+                projectId,
+                state: {
+                    type: cloudService.ProjectState.opened,
+                },
+            })
+        }
+    }
+
+    async openProject(projectId: cloudService.ProjectId): Promise<void> {
+        this.currentlyOpeningProjectId = projectId
+        const project = await this.projectManager.openProject({
+            projectId,
+            missingComponentAction: projectManager.MissingComponentAction.install,
+        })
+        this.currentlyOpenProject = { id: projectId, project }
+    }
+}
+
+// =====================
+// === createBackend ===
+// =====================
+
+/** Shorthand method for creating a new instance of the backend API. */
```

Comment:
1. Why `createBackend()` is "shorthand" for `new Backend()`, while the latter is in fact shorter?
2. The longer I read this file the more I believe the naming in this file is bad. Are we really creating new backend here? I think we are not spawning backend here.

Reply:
this file was derived from `cloudService.ts`, but that's a good point. the "backend" in this case is the project manager, and it's creating a new *api wrapper* rather than a backend, but it's definitely more than doable to improve naming all around